src/modules/editor/reducer.ts

```diff
@@ -217,6 +235,32 @@ export const editorReducer = (state = INITIAL_STATE, action: EditorReducerAction
         multiselectionEnabled: action.payload.enabled
       }
     }
+    case SET_BODY_SHAPE: {
+      return {
+        ...state,
+        bodyShape: action.payload.bodyShape
+      }
+    }
+    case SET_AVATAR_ANIMATION: {
+      return {
+        ...state,
+        avatarAnimation: action.payload.animation
+      }
+    }
+    case SET_ITEMS: {
+      return {
+        ...state,
+        visibleItemIds: action.payload.items
+          .filter(item => item.data.representations.some(r => r.bodyShape.includes(state.bodyShape))) // only add items that have a valid representation for the selected body shape
```

Comment:
I think the same logic is already present on `LeftPanel/Items/Items.tsx`, maybe it's a good idea to extract it?

Reply:
done 👍